docker-app/qfieldcloud/core/models.py

```diff
@@ -1036,10 +1032,25 @@ class Meta:
             "If enabled, QFieldCloud will automatically overwrite conflicts in this project. Disabling this will force the project manager to manually resolve all the conflicts."
         ),
     )
+
     thumbnail_uri = models.CharField(
         _("Thumbnail Picture URI"), max_length=255, blank=True
     )
 
+    storage_keep_versions = models.PositiveIntegerField(
+        default=10,
+        help_text=(
+            "If enabled, QFieldCloud will use this value to limit the maximum number of versions per file in the current project with this value. If the value is larger than the maximum number of versions per file your current plan entitles you to, the current plan's value will be used instead."
+        ),
+    )
+
+    use_storage_keep_versions = models.BooleanField(
+        _("Opt-in to project-based max. files versions"),
+        default=False,
+        null=False,
+        blank=False,
+    )
+
```

Comment:
Can't we just avoid this field here? If a new project is created, then it automatically get's the owner current subscription plan `storage_keep_versions` value and it is set. Then the user can change it to whatever they want within the allowed values.

Reply:
How does that work when the user wants to opt-out from this and just use the max. number of versions offered by their plan?